### PR TITLE
Send Amount and Token Form

### PR DESCRIPTION
**Pull request by youngkidwarrior**

### CODE REVIEW

**Comment by youngkidwarrior** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/0xsend/sendapp/383?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#383** <a href="https://app.graphite.dev/github/pr/0xsend/sendapp/383?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#382** <a href="https://app.graphite.dev/github/pr/0xsend/sendapp/382?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#381** <a href="https://app.graphite.dev/github/pr/0xsend/sendapp/381?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#380** <a href="https://app.graphite.dev/github/pr/0xsend/sendapp/380?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#379** <a href="https://app.graphite.dev/github/pr/0xsend/sendapp/379?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `dev`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @youngkidwarrior and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/app/features/send/SendAmountForm.tsx [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+
+  const { setParams } = useSendParams()
+  const {
+    params: { amount: amountParam, sendToken: tokenParam, recipient: recipientParam },
+  } = useSendParams()
+
+  const token = form.watch('token')
+
+  useEffect(() => {
+    const subscription = form.watch(({ amount, token }) => {
```
Comment (0xBigBoss) [inline packages/app/features/send/SendAmountForm.tsx:36]: sonice

File: packages/app/features/send/SendAmountForm.tsx [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+      setParams(
+        {
+          amount: amount?.toString(),
+          sendToken: token as `0x${string}` | 'eth',
```
Comment (0xBigBoss) [inline packages/app/features/send/SendAmountForm.tsx:40]: just a nit i would just make a type for this `sendToken` or even just `token` 

```ts
type token = `0x${string}` | 'eth'
```

**Review by 0xBigBoss — APPROVED** [review]
lgtm

File: packages/app/features/send/SendAmountForm.tsx (hunk 2 above)
Comment (youngkidwarrior) [reply packages/app/features/send/SendAmountForm.tsx:40]: Yeah, right now we have the `token` and `sendToken` params

I'm going to make them the same param in the future, but `token` has a dependency in TopNav. Once I refactor TopNav, I'll clean this up

**Comment by youngkidwarrior** [comment]
### Merge activity

* **May 11, 5:36 AM EDT**: @youngkidwarrior started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/0xsend/sendapp/383).

**Pull request merged**